packages/amazonq/src/extension.ts

```diff
@@ -149,10 +149,16 @@ export async function activateAmazonQCommon(context: vscode.ExtensionContext, is
         }
 
         const authState = (await AuthUtil.instance.getChatAuthState()).codewhispererChat
+        const currConn = AuthUtil.instance.conn as SsoConnection
```

Comment:
I'm not sure in what cases but `.conn` can be an IAM connection, see `AuthUtil.getCredentials()`.

Look to return early or throw if `isAnySsoConnection()` is false. This should also automatically cast the conn to an SsoConnection for you